lib/backend/firestore/firestorebk_test.go

```diff
@@ -261,6 +257,14 @@ func TestDeleteDocuments(t *testing.T) {
 					CreateTime: time.Now(),
 					UpdateTime: time.Now(),
 				})
+
+				// We really shouldn't need this, but the Firestore SDK made some unfortunate design
+				// decisions that make it impossible to set the field of a DocumentRef used for the seemingly
+				// useless deduplication in the BulkWriter API.
+				rs := reflect.ValueOf(docs[i].Ref).Elem()
+				rf := rs.FieldByName("shortPath")
+				rf = reflect.NewAt(rf.Type(), unsafe.Pointer(rf.UnsafeAddr())).Elem()
+				rf.SetString(docs[i].Ref.Path)
```

Comment:
This was already merged due to @rosstimothy adding it to the queue. I was confused by errors for some time in tests for quite some time that didn't crop up in real world testing. As it turns out, the firestore SDK `BulkWriter` API we now use assembles batches of changes internally and submits them. Each batch is deduplicated so you can't get two deletes or an edit after a delete in the same batch, so they always succeed. The consequence of this is that it's illegal to delete a document twice with the same BulkWriter.

This would be fine, the test generates an unique `ref.Path` right? Well, as it turns out, the SDK does deduplication on an internal hidden variable which represents the path in a smaller encoding. Why? I have no idea. But this is stored in a hidden field that we cannot set easily. If you remove this, you get deduplication errors in the tests, even if the `ref.Path` is unique.

Reply:
I could not find an alternative way around this easily.

Reply:
Fair enough, thanks for the explanation.